Hidden files are incorrectly picked as CloudFormation templates

I have run into the issue below when running CF seed on my laptop.

```
INFO -------- Collecting templates not included in the environment ---------
INFO Loading template for stack .DS_Store from cloudformation/.DS_Store...
ERROR 'utf-8' codec can't decode byte 0x80 in position 3131: invalid start byte
ERROR Aborting deployment
make[1]: *** [deploy] Error 8
make: *** [appd] Error 2
```

--- a/.gitignore
+++ b/.gitignore
@@ -10,3 +10,4 @@ dist/**
 **/node_modules
 .python-version
 *.patch
+/tmp

--- a/README.md
+++ b/README.md
@@ -63,7 +63,7 @@ Quick start
 1. Create a new directory for your project
 2. Copy everything from the `examples` directory to the root of the project
 3. Edit `parameters/dev.yaml` to your needs
-4. Add more stuff under the `cloudformation` directory and include it in `parameters/dev.yaml`
+4. Add more templates with `.cf.yaml` extensions under the `cloudformation` directory and include them in `parameters/dev.yaml`
 
 ### Finally:
 
@@ -125,10 +125,10 @@ You can also tag your stacks/stacksets by defining your tags as a dictionary and
 tags_a: &TAGSA
   testkey1: testvalue1
   testkey2: testvalue2
-  
+
 tags_b: &TAGSB
   testkey3: testvalue3
-  
+
   stacks:
   - name: example-stackset-template
     type: stackset

--- a/cloudformation_seed/cfn_template.py
+++ b/cloudformation_seed/cfn_template.py
@@ -90,7 +90,7 @@ class CloudformationCollection(util.DirectoryScanner):
                     environment_parameters: Dict['str', Any]) -> None:
         self.s3_bucket: Any = s3_bucket
         self.environment_parameters: Dict['str', Any] = environment_parameters
-        self.template_files: List[Tuple[str, str]] = self.scan_directories(path, '*.cf.yaml')
+        self.template_files: List[Tuple[str, str]] = self.scan_directories(path, '**/*.cf.yaml')
         util.log_section('Collecting templates included in the environment')
         self.templates: List[CloudformationTemplate] = [
             CloudformationTemplate(

--- a/cloudformation_seed/cfn_template.py
+++ b/cloudformation_seed/cfn_template.py
@@ -90,7 +90,7 @@ class CloudformationCollection(util.DirectoryScanner):
                     environment_parameters: Dict['str', Any]) -> None:
         self.s3_bucket: Any = s3_bucket
         self.environment_parameters: Dict['str', Any] = environment_parameters
-        self.template_files: List[Tuple[str, str]] = self.scan_directories(path)
+        self.template_files: List[Tuple[str, str]] = self.scan_directories(path, '*.cf.yaml')
         util.log_section('Collecting templates included in the environment')
         self.templates: List[CloudformationTemplate] = [
             CloudformationTemplate(

--- a/cloudformation_seed/util.py
+++ b/cloudformation_seed/util.py
@@ -60,11 +60,13 @@ ORG_ARN_RE = re.compile(r'^arn:aws:organizations::\d{12}:\w+/(?P<org_id>o-\w+)')
 
 
 class DirectoryScanner(object):
-    def scan_directories(self, path: str, glob: str = '*') -> List[Tuple[str, str]]:
+    def scan_directories(self, path: str, glob: str = '**/*') -> List[Tuple[str, str]]:
         u = list()
         for item in Path(path).glob(glob):
             if Path.is_file(item):
-                u.extend([(item.name, str(item))])
+                relative_path = str(item)
+                key = relative_path[len(path):].strip(os.sep)
+                u.extend([(key, relative_path)])
         return u
 
 

--- a/cloudformation_seed/util.py
+++ b/cloudformation_seed/util.py
@@ -1,5 +1,5 @@
 from typing import List, Tuple, Dict, Any, Optional, Union
-
+from pathlib import Path
 import logging
 import boto3
 import re
@@ -60,11 +60,11 @@ ORG_ARN_RE = re.compile(r'^arn:aws:organizations::\d{12}:\w+/(?P<org_id>o-\w+)')
 
 
 class DirectoryScanner(object):
-    def scan_directories(self, path: str) -> List[Tuple[str, str]]:
+    def scan_directories(self, path: str, glob: str = '*') -> List[Tuple[str, str]]:
         u = list()
-        for root, _, files in os.walk(path):
-            relative_root = root.replace(path, '').strip(os.sep)
-            u.extend([(f'{relative_root}/{f}'.replace(os.sep, '/').strip('/'), os.path.join(root, f)) for f in files])
+        for item in Path(path).glob(glob):
+            if Path.is_file(item):
+                u.extend([(item.name, str(item))])
         return u
 
 

